pkg/controller/node/node_controller.go

```diff
@@ -157,7 +158,7 @@ func listAttachmentsByNodeExternalId(nsxClients *NsxClients, vmExternalId string
 	}
 	for _, vif := range vifs.Results {
 		if vif.LportAttachmentId != "" {
-			attachment_ids = append(attachment_ids, vif.LportAttachmentId)
+			attachment_ids.Insert(vif.LportAttachmentId)
```

Comment:
Current solution certainly can fix duplicated LportAttachmentId case. 
however,from search output, we see there two vis associated with the same node. I see one has invalid IpAddressInfo and other has empty IpAddressInfo. I think the latter one with empty IpAddressInfo is from power-offed vm. So, do we need to check which vis is valid using IpAddressInfo  info.

Reply:
It's safe because we don't read the address from the vifs, we read it from K8s side. Our target is to get the node LSP id. The 2 vifs have the same attachment id, and there is only one port refering to the attachment id. In the other words, the vif attachment and the logic port is unique, that's enough. The logic flow is: get IP and provider id from K8s side -> get node external id by provider id -> get attachment ids by node external id -> get logic ports by attachment ids -> filter the logic ports by IP, then the result should be a single node LSP.

Reply:
Thanks for detailed depiction. It should be ok to skip IpAddressInfo check.